File: app/src/main/java/com/chaman/app/MyApplication.java
package com.chaman.app;

import com.chaman.android.base.BaseApplication;
import com.chaman.app.dagger.component.AppComponent;
import com.chaman.app.dagger.component.DaggerAppComponent;
import com.chaman.app.dagger.module.CacheModule;
import com.chaman.app.dagger.module.ServiceModule;

/**
 * ****************************************************************
 * Author: Chaman
 * Date: 2017/5/23 下午12:58
 * Desc:
 * *****************************************************************
 */

public class MyApplication extends BaseApplication{
    private static AppComponent mAppComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        mAppComponent = DaggerAppComponent.builder()
                .appModule(getAppModule())
                .clientModule(getClientModule())
                .serviceModule(new ServiceModule())
                .cacheModule(new CacheModule())
                .build();
    }

    @Override
    protected String getBaseUrl() {
        return "http://api.link-pay.cn/";
    }
    /**
     * 将AppComponent返回出去,供其它地方使用, AppComponent接口中声明的方法返回的实例, 在getAppComponent()拿到对象后都可以直接使用
     *
     * @return
     */
    public  AppComponent getAppComponent() {
        return mAppComponent;
    }


}

File: demo/src/main/java/com/lcm/app/MyApplication.kt
package com.lcm.app

import com.alibaba.android.arouter.launcher.ARouter
import com.lcm.android.base.BaseApplication
import com.lcm.app.dagger.component.AppComponent
import com.lcm.app.dagger.component.DaggerAppComponent
import com.lcm.app.dagger.module.DataModule
import com.squareup.leakcanary.LeakCanary
import io.realm.Realm
import io.realm.RealmConfiguration

/**
 * ****************************************************************
 * Author: LCM
 * Date: 2018/5/4 11:49
 * Desc:
 * *****************************************************************
 */
class MyApplication:BaseApplication(){

    var appComponent: AppComponent?=null
        private set

    override fun onCreate() {
        super.onCreate()
        appComponent = DaggerAppComponent.builder()
                .appModule(mAppModule)
                .clientModule(mClientModule)
                .dataModule(DataModule())
                .build()


        ARouter.openDebug()
        ARouter.openLog()
        ARouter.init(this)

        Realm.init(this)
        Realm.setDefaultConfiguration(RealmConfiguration.Builder().name(BuildConfig.DB_Name).build())

        LeakCanary.install(this)
    }



}